AppWidget/glance-widget/src/main/java/com/example/android/glancewidget/TodoListGlanceWidget.kt

```diff
@@ -0,0 +1,157 @@
+import androidx.compose.ui.unit.sp
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.glance.GlanceId
+import androidx.glance.GlanceModifier
+import androidx.glance.LocalContext
+import androidx.glance.action.ActionParameters
+import androidx.glance.action.actionParametersOf
+import androidx.glance.appwidget.CheckBox
+import androidx.glance.appwidget.GlanceAppWidget
+import androidx.glance.appwidget.GlanceAppWidgetReceiver
+import androidx.glance.appwidget.action.ActionCallback
+import androidx.glance.appwidget.action.actionRunCallback
+import androidx.glance.appwidget.lazy.LazyColumn
+import androidx.glance.appwidget.lazy.items
+import androidx.glance.appwidget.state.updateAppWidgetState
+import androidx.glance.background
+import androidx.glance.currentState
+import androidx.glance.layout.Column
+import androidx.glance.layout.fillMaxSize
+import androidx.glance.layout.fillMaxWidth
+import androidx.glance.layout.padding
+import androidx.glance.state.GlanceStateDefinition
+import androidx.glance.state.PreferencesGlanceStateDefinition
+import androidx.glance.text.FontWeight
+import androidx.glance.text.Text
+import androidx.glance.text.TextStyle
+
+private val groceryStringIds = listOf(
+    R.string.grocery_list_milk,
+    R.string.grocery_list_eggs,
+    R.string.grocery_list_tomatoes,
+    R.string.grocery_list_bacon,
+    R.string.grocery_list_butter,
+    R.string.grocery_list_cheese,
+    R.string.grocery_list_potatoes,
+    R.string.grocery_list_broccoli,
+    R.string.grocery_list_salmon,
+    R.string.grocery_list_yogurt
+)
+
+/**
+ * Glance widget that showcases how to use:
+ * - Compound buttons
+ * - LazyColumn
+ * - State management using GlanceStateDefinition
+ */
+class TodoListGlanceWidget : GlanceAppWidget() {
+
+    override var stateDefinition: GlanceStateDefinition<*> = PreferencesGlanceStateDefinition
+
+    @Composable
+    override fun Content() {
+        Column(
+            modifier = GlanceModifier
+                .fillMaxSize()
+                .background(color = Color.White)
+                .padding(8.dp)
+        ) {
+            val context = LocalContext.current
+            val prefs = currentState<Preferences>()
+            Text(
+                text = context.getString(R.string.glance_todo_list),
+                modifier = GlanceModifier
+                    .fillMaxWidth()
+                    .padding(8.dp),
+                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 18.sp),
+            )
+            CountToggled()
+            LazyColumn {
+                items(groceryStringIds) {
+                    val toggledString = context.getString(it)
+                    val checked = prefs[booleanPreferencesKey(toggledString)] ?: false
+                    CheckBox(
+                        text = toggledString,
+                        checked = prefs[booleanPreferencesKey(toggledString)] ?: false,
```

Comment:
Why not re-used the checked variable you created just above?

Reply:
Sorry just overlooked I extracted it to a variable, done.